ui/src/app/kvm/views/KVMDetails/CoreResources/CoreResources.tsx

```diff
@@ -1,16 +1,28 @@
 import classNames from "classnames";
 
 import PodMeter from "app/kvm/components/PodMeter";
+import { getRanges } from "app/utils";
 
 export type Props = {
   cores: {
     allocated: number;
     free: number;
   };
   dynamicLayout?: boolean;
+  pinned?: number[];
+  available?: number[];
 };
 
-const CoreResources = ({ cores, dynamicLayout }: Props): JSX.Element => {
+const CoreResources = ({
+  cores,
+  dynamicLayout,
+  pinned = [],
+  available = [],
+}: Props): JSX.Element => {
+  const noPin = (pinned && pinned.length === 0) || 0;
+  const pinnedRanges = getRanges(pinned).join(",");
+  const noAvailable = (available && available.length === 0) || 0;
+  const availableRanges = getRanges(available).join(",");
```

Comment:
Do you think having a space between the cores is more readable? `1, 2, 3` instead of `1,2,3`. In that case you would use `join(", ")`